src/cryptonote_core/service_node_list.cpp

```diff
@@ -1337,7 +1337,13 @@ namespace service_nodes
     m_height = hardfork_9_from_height;
   }
 
-  bool convert_registration_args(cryptonote::network_type nettype, std::vector<std::string> args, std::vector<cryptonote::account_public_address>& addresses, std::vector<uint64_t>& portions, uint64_t& portions_for_operator, bool& autostake)
+  bool convert_registration_args(cryptonote::network_type nettype,
+                                 std::vector<std::string> args,
+                                 std::vector<cryptonote::account_public_address>& addresses,
+                                 std::vector<uint64_t>& portions,
+                                 uint64_t& portions_for_operator,
+                                 bool& autostake,
+                                 boost::optional<std::string&> err_msg)
```

Comment:
My c++ foo is not strong here but if its ```boost::optional<std::string &> err_msg```  and you do

```
if (err_msg) *err_msg = msg;
```

Then it ends up std::moving or allocating new memory and copying it into the given string, so although you give the callee the ability to specify their string, say a pre existing error message to append to or pre allocated memory, it doesn't get used?

Reply:
I misunderstood. This is all good.